examples/call_multiple_tools_directly_with_auth.py

```diff
@@ -0,0 +1,178 @@
+    )
+
+    # Step 2.5: Use the previous tool output to construct the inputs for the following tool calls
+    audio_features = response["audio_features"][0]
+    recommendation_params.update({
+        "target_acousticness": float(audio_features["acousticness"]),
+        "target_danceability": float(audio_features["danceability"]),
+        "target_energy": float(audio_features["energy"]),
+        "target_instrumentalness": float(audio_features["instrumentalness"]),
+        "target_key": int(audio_features["key"]),
+        "target_liveness": float(audio_features["liveness"]),
+        "target_loudness": float(audio_features["loudness"]),
+        "target_mode": int(audio_features["mode"]),
+        "target_speechiness": float(audio_features["speechiness"]),
+        "target_tempo": float(audio_features["tempo"]),
+        "target_time_signature": int(audio_features["time_signature"]),
+        "target_valence": float(audio_features["valence"]),
+    })
+
+    # Step 3: Get song recommendations that are similar to the currently playingsong
+    print(
+        f"Getting recommendations similar to '{current_track_name}' by {', '.join(current_track_artists)} - {current_track_spotify_url}"
+    )
+    response = call_tool(
+        client, get_recommendations_tool, user_id, inputs=json.dumps(recommendation_params)
+    )
+
+    # Step 3.5: Use the previous tool output to construct the inputs for the following tool calls
+    # Filter out remixes and the same song from the recommendations
+    tracks = [
+        track for track in response["tracks"] if not track["name"].startswith(current_track_name)
+    ]
+    track_ids = [
+        track["id"] for track in tracks if (user_country_code in track["available_markets"])
+    ]
+    track_names = [track["name"] for track in tracks]
+    tracks_artists = [[artist["name"] for artist in track["artists"]] for track in tracks]
+    track_spotify_urls = [track["external_urls"]["spotify"] for track in tracks]
+
+    if not track_ids:
+        print("I couldn't find any similar songs that are available in your country.")
+        return
+
+    print("\nHere are some recommendations:")
+    for track_name, track_artists, track_spotify_url in zip(
+        track_names, tracks_artists, track_spotify_urls
+    ):
+        print(f"\t{track_name} by {', '.join(track_artists)} - {track_spotify_url}")
+
+    # Step 4: Start playing the recommended songs
+    response = call_tool(
+        client,
+        start_tracks_playback_tool,
+        user_id,
+        inputs=json.dumps({"track_ids": track_ids}),
+    )
+
+    # Step 5: Inform the user which recommended song is now playing
+    response = call_tool(client, get_currently_playing_tool, user_id)
+    print(
+        f"\nNow playing: {response['track_name']} by {', '.join(response['track_artists'])} - {response['track_spotify_url']}"
+    )
+
+
+if __name__ == "__main__":
+    client = Arcade(base_url="https://api.arcade-ai.com")
```

Comment:
Since the new Spotify tools haven't been deployed to production yet, this won't work if you try it out. If you would like to try out this example, then please use localhost

Reply:
Should be good now right?

Reply:
not yet. The minor changes to the spotify toolkit in this PR will need to be deployed for this to work with cloud engine

Reply:
Ah, gotcha